src/components/Missions/MissionsPage.js

```diff
@@ -1,16 +1,32 @@
 const MissionsPage = (prop) => {
-  const { name, description } = prop;
+  const {
+    joinMissionHandler, id, name, description, reserved, leaveMissionHandler,
+  } = prop;
+
+  let button;
+  if (!reserved) {
+    button = <button type="button" className="mission-button-join" onClick={() => joinMissionHandler(id)}>Join Mission</button>;
+  } else {
+    button = <button type="button" className="mission-button-badge" onClick={() => leaveMissionHandler(id)}>Leave Mission</button>;
+  }
+
+  let badge;
+  if (!reserved) {
+    badge = <span className="mission-button">Not a Member</span>;
+  } else {
+    badge = <span className="join-badge ">Active Member</span>;
+  }
 
   return (
 
     <tr>
       <td className="mission-title">{name}</td>
       <td className="mission-description">{description}</td>
       <td>
-        <button className="mission-button" type="button">Not a Member</button>
+        {badge}
       </td>
       <td>
-        <button className="mission-button-join" type="button">Join Mission</button>
+        {button}
```

Comment:
To render the button if you do the function you only have to call the function instead of the variable something like this:

> { RenderButton(reserved) }